packages/memory/readme.md

```diff
@@ -0,0 +1,60 @@
+# `@directus/memory`
+
+Directus has various different needs for ephemeral storage that's synced between multiple processes for the same
+Directus Projects. To streamline that setup, this package exports three classes that are used for everything related to
+ephemeral storage:
+
+- [Cache](#cache)
+- [Bus](#bus)
+- [Limiter](#limiter)
+
+## Cache
+
+The cache class is a LRU-based key-value store.
```

Comment:
The Redis side is based on how you have Redis configured.. I wanna rely on Redis' native LRU cache capabilities which is handled on the redis config side rather than the per-connection or command side of things. We'll have to rely on documentation for self-hosting people to make sure to be aware (hence the revived production checklist-discussion earlier today @phazonoverload 🙂 )

Reply:
TIL that Redis has native LRU cache capabilities 😅 When I was researching how to implement a proper LRU cache on top of Redis quite some time ago, I didn't come across this.

This does mean however, that we'd have to encourage users to use different Redis instances for cache vs storage and maybe even different instances for different types of caches (e.g. one for schema cache and one for request cache), right?

Reply:
> This does mean however, that we'd have to encourage users to use different Redis instances for cache vs storage and maybe even different instances for different types of caches (e.g. one for schema cache and one for request cache), right?

This is where things get... interesting. Unfortunately the LRU config for Redis is global, so you _have_ to use multiple Redis instances if you want different LRU configs for different purposes. pub/sub messages are excluded as they're not stored at all. I really wanna try figuring out a way we can rely on the same Redis instance across Directus. Needing two instances really makes hosting and scaling more complicated than I want (and re-implementing LRU logic in JS if Redis already has it highly optimized feels silly).

I'm hopeful we can update the multi-stage cache to use the pubsub bus to invalidate the local memory, and then have both LRU on local and Redis memory for both schema and data in the same cache object. Seeing it's LRU, the schema information shouldn't be invalidated as much as it's often accessed 🤞🏻